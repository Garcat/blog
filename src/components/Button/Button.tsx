import { FC, HTMLProps, ReactNode } from 'react'
import classNames from 'classnames'

export enum Variant {
  Primary,
  Secondary,
  Dark,
  Outline,
  Background,
}

export interface Props extends HTMLProps<HTMLButtonElement> {
  variant?: Variant
  block?: boolean
  children: ReactNode
  type?: 'submit' | 'button' | 'reset' | undefined
}

const Button: FC<Props> = ({
  variant = Variant.Primary,
  block = false,
  children,
  type = 'button',
  ...props
}) => {
  const classname = classNames(
    'h-12',
    'rounded-md',
    'px-4',
    'disabled:opacity-50',
    'disabled:grayscale',
    'disabled:cursor-not-allowed',
    variant === Variant.Primary && ['bg-background', 'text-primary'],
    variant === Variant.Secondary && ['bg-primary', 'text-on-primary'],
    variant === Variant.Background && ['bg-background', 'text-dark'],
    variant === Variant.Dark && ['bg-dark', 'text-background'],
    variant === Variant.Outline && ['border', 'border-outline', 'text-on-background'],
    block && ['block', 'w-full']
  )
  return (
    // eslint-disable-next-line react/button-has-type, react/jsx-props-no-spreading
    <button type={type} className={classname} {...props}>
      {children}
    </button>
  )
}

export default Button
