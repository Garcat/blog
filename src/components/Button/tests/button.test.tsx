/* eslint-disable testing-library/await-async-query */
import { render, screen } from '@testing-library/react'
import Button from '../Button'

describe('Button test', () => {
  it('renders test with default setting', () => {
    const view = render(<Button>Test</Button>)
    expect(screen.findByRole('button', { name: 'Test' })).toBeTruthy()
    expect(view).toMatchSnapshot()
  })
})
