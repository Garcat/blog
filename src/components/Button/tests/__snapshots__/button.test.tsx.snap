// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test renders test with default setting 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="disabled:opacity-20 disabled:grayscale disabled:cursor-not-allowed rounded-md px-4 bg-background text-on-background h-12"
        type="button"
      >
        Test
         
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="disabled:opacity-20 disabled:grayscale disabled:cursor-not-allowed rounded-md px-4 bg-background text-on-background h-12"
      type="button"
    >
      Test
       
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
